packages/application-shell/src/components/navbar/use-navbar-state-manager.ts

```diff
@@ -34,10 +34,30 @@ type Action =
   | { type: 'setIsMenuOpenAndMakeExpanderVisible'; payload: boolean }
   | { type: 'reset' };
 
+const cachedIsForcedMenuOpen = window.localStorage.getItem(
+  STORAGE_KEYS.IS_FORCED_MENU_OPEN
+);
+
+const getShouldInitiallyOpen = (): boolean => {
+  if (!isNil(cachedIsForcedMenuOpen)) {
+    return JSON.parse(cachedIsForcedMenuOpen);
+  }
+  return false;
+};
+
+const shouldBeInitiallyOpen: boolean = getShouldInitiallyOpen();
+
+// GIVEN user has `STORAGE_KEYS.IS_FORCED_MENU_OPEN=true`
+// THEN update the DOM immediately and not defer that to the reducer
+// this is done to avoid the flickering experience of the menu once the user loads the application
+if (shouldBeInitiallyOpen) {
+  document.body.classList.add('body__menu-open');
+}
 const initialState = {
   isExpanderVisible: true,
-  isMenuOpen: false,
+  isMenuOpen: shouldBeInitiallyOpen,
```

Comment:
I wouldn't define it "statically" like this.

Can we maybe change this into a function `getInitialState` and pass the `isMenuOpen` to the function?

Then you do

```js
useReducer(reducer, getInitialState(cachedIsForcedMenuOpen))
```

```

Reply:
> Thanks for looking into this. I have an alternative proposal though...

I am fine with the proposal with the initial state not defined statically. With this, we can keep `cachedIsForcedMenuOpen` scoped inside the hook.

---

> ...to avoid using window and document when the module loads.

I've already tried this, since this was my initial approach

However, avoiding to use window and document when the module loads brings us back to the initial problem where updating the list to be expanded immediate is deferred to the reducer.

Reply:
basically, we need to keep this

```js
if (shouldBeInitiallyOpen) {
  document.body.classList.add('body__menu-open');
}
```

Reply:
I get that. What I'm suggesting is to pass the cached value to the `initialState` of the reducer. I haven't tried this but I assume it should work.

Reply:
okay, sure. We do that already.

https://github.com/commercetools/merchant-center-application-kit/blob/ef314ecc6fb4388e349479cb3c425faac500d649/packages/application-shell/src/components/navbar/use-navbar-state-manager.ts#L147:L147

but if we want to do this via a function, sure. we can do that too. I'll come back with an update later

Reply:
maybe something in this direction?
f7be21b